### ISSUE TITLE
Created GUI to visualize transformations

### ISSUE THREAD

**Issue by phrasenmaeher** [iver56/audiomentations#104]
Hi @iver56,

I have used your `audiomentations` library for my research and found it quite useful and easy to use- thanks for the many provided transformations!

To see (and hear!) the effects of different transformations, I have created a small web app. I have used `streamlit' and your library; one can upload a (wave) file, select different augmentations, and then examine the results. 

I have mainly used this for my own research, but other users might find it helpful to interactively explore the effects of different augmentation techniques! You can find the code [here](https://github.com/phrasenmaeher/audio-transformation-visualization) and the running app (generously hosted by streamlit sharing) [here](https://share.streamlit.io/phrasenmaeher/audio-transformation-visualization/main/visualize_transformation.py).

**Comment by iver56** [comment]
That's cool - good initiative :) I've been thinking of making something like that myself.

Would you like this tool to become something more, like an official interactive audiomentations demo?

A little feedback: I tried it on a 44.1 kHz wav file and it sounded very pitched down and stretched. Maybe your tool assumed a different sample rate, like 16 kHz? Ideally it could read the sample rate from the file and use that

**Comment by phrasenmaeher** [comment]
Thank you for the kind feedback!

Yes, I am interested; an official interactive demo sounds good to me!

I'll look into the mentioned issue; currently it can only handle wav files (but should do so at any rate).

**Comment by iver56** [comment]
Here is the file (zipped) that I tried:
[state.zip](https://github.com/iver56/audiomentations/files/6630296/state.zip)

**Comment by iver56** [comment]
Some things I have in mind right now:

* Would you want to implement support for visualizing a pipeline/`Compose` of transforms? E.g. A compose that does pitch shifting first and then time stretching
* How will it deal with transforms that need external files, like AddBackgroundNoise?

**Comment by iver56** [comment]
It would also be nice to allow users to use a demo file at the click of a button. Not all users have a suitable wav file readily available :)

**Comment by iver56** [comment]
@phrasenmaeher Does this look OK to you? https://github.com/iver56/audiomentations/pull/105

**Comment by phrasenmaeher** [comment]
Hi @iver56, this is totally fine for me!

Regarding the demo file: I'll include some freely available files, probably from the [ESC50 dataset](https://github.com/karolpiczak/ESC-50). 

For any transformations that require external files, I'll include an option to either upload a separate one or use one that comes shipped.

**Comment by iver56** [comment]
Nice. Looking forward to that.

**Comment by phrasenmaeher** [comment]
> Some things I have in mind right now:
> 
> * Would you want to implement support for visualizing a pipeline/`Compose` of transforms? E.g. A compose that does pitch shifting first and then time stretching
> * How will it deal with transforms like AddBackgroundNoise, that need external files?

I am afraid that I don't understand your first point, could you please clarify that?

If you mean that I visualize the pipeline, then yes, that's actually what I am doing rate now. I'll include this into the documentation so this is clearly stated. I think you thought that it visualizes the transformations individually; but it runs the steps of the pipeline individually. This is where it happens, the output of the former step is the input to the next step:
```
    for col_index, individual_transformation in enumerate(pipeline.transforms):
        transformation_name = str(type(individual_transformation)).split("'")[1].split(".")[-1]
        modified = individual_transformation(y, sr)
        fig = plot_transformation(modified, sr, transformation_name=transformation_name)
        y = modified
```

**Comment by iver56** [comment]
Ok, thanks for the clarification

**Comment by phrasenmaeher** [comment]
I have updated the application, you can see the current version running here: 

https://share.streamlit.io/phrasenmaeher/audio-transformation-visualization/main/visualize_transformation.py

**Comment by iver56** [comment]
Thanks :) That's better 👍 

Btw, I found something that looks like a bug: The sound in the last step became silent. Not sure if this bug is in audiomentations or your GUI tool. Steps to reproduce:
* Select the dog example audio as input
* Check "Gain" and "ClippingDistortion"
* Hit Apply

![bilde](https://user-images.githubusercontent.com/1470603/121778726-7eef1f80-cb98-11eb-99d1-62222b3b5d0f.png)

**Comment by phrasenmaeher** [comment]
I could reproduce this and traced it back to the parameters of the ClippingDistortion. I lowered the `max_percentile_threshold` to 10 to fix it.

**Comment by atamazian** [comment]
The app is no longer available.

**Comment by phrasenmaeher** [comment]
Hi @atamazian, thanks for the note. 
I guess that is related to the latest update in streamlit -- I'll wait a day or two and then check again. Thanks for your patience.

**Comment by atamazian** [comment]
Great! How about adding some new transforms from v0.17.0?

**Comment by phrasenmaeher** [comment]
v0.17.0 introduced the ``Clip`` transformation, which I have added. Currently, only audio transformations are supported, not spectrogram transformations.

**Comment by iver56** [comment]
v0.18.0 (released today) added four new transforms: BandPassFilter, HighPassFilter, LowPassFilter and Reverse

**Comment by phrasenmaeher** [comment]
Thanks for the notification! I'll update the application as of tomorrow!

**Comment by atamazian** [comment]
Is it possible to change transform settings?

**Comment by phrasenmaeher** [comment]
No, currently not. Almost all transformation settings are on their default value--with the exception of the probability, ``p``, which I've consistently set to ``1.0`` to reliably transform the input.

However, you can clone the project locally and adapt the code, which would be the fastest way to play with the paramters.

**Comment by atamazian** [comment]
I see, thanks for replying.

**Comment by iver56** [comment]
v0.21.0 was released today :) It has some new transforms. Wanna update?

**Comment by phrasenmaeher** [comment]
Sure, thanks for letting me know! 

(It will take some time, think two weeks. I'll respond here; if I don't then remind me here)

**Comment by phrasenmaeher** [comment]
I've updated the script to support the four novel transformations ([commit](https://github.com/phrasenmaeher/audio-transformation-visualization/commit/59c892ccd496fd3eeb8c228a7f46ccc66c9cab8c)).

**Comment by iver56** [comment]
Sweet, thanks :) You could also add the new `GainTransition` in the next update

**Comment by phrasenmaeher** [comment]
> More transforms have been added. Can you update the app?

I have updated the application with the new transformations ([commit](https://github.com/phrasenmaeher/audio-transformation-visualization/commit/a1b9a2d1ff4819ecb3607371de31b53a5f58ceb5)).

May I ask that on the documentation page, you add a link to my GitHub profile (and not merely the name)? Thanks!

**Comment by iver56** [comment]
Thanks!

[Done](https://github.com/iver56/audiomentations/commit/c595efdca7281c3da78ad507d73aff5a1d11861c)
